helm/hpcc/templates/_warnings.tpl

```diff
@@ -25,13 +25,13 @@
 Generate a list of warnings
 Pass in dict with root with warnings
 */}}
-{{- define "hpcc.getWarnings" -}}
+{{ define "hpcc.getWarnings" }}
  {{- $warnings := list -}}
  {{- $defaultVersion := .root.Values.global.image.version | default .root.Chart.Version -}}
  {{- if (ne $defaultVersion .root.Chart.Version) -}}
   {{- $warning := dict "source" "helm" "severity" "warning" -}}
   {{- $_ := set $warning "msg" (printf "The expected image version for this helm chart is %s" .root.Chart.Version) -}}
-  {{- $warnings := append $warnings $warning -}}
+  {{- $warnings = append $warnings $warning -}}
```

Comment:
I'm not keen on this. It's undocumented behaviour afaics (and the sole use of this).
As elsewhere, if we need to set something inside a scope, we use set it into a dictionary with $_ : = set ...

Reply:
I'll try to get it to work with :=